driver/web/docs/resources/api/user/coursesKey.yaml

```diff
@@ -0,0 +1,101 @@
+get:
+  tags:
+  - Client
+  summary: Get user course
+  description: |
+    Get user course
+  security:
+    - bearerAuth: []
+  parameters:
+    - name: key
+      in: path
+      description: course key
+      required: true
+      style: simple
+      explode: false
+      schema:
+        type: string
+  responses:
+    200:
+      description: Success
+      content:
+        application/json:
+          schema:
+            $ref: "../../../schemas/custom/UserCourse.yaml"
+    400:
+      description: Bad request
+    401:
+      description: Unauthorized
+    500:
+      description: Internal error
+  x-core-function: GetUserCourse
+  x-data-type: model.UserCourse
+  x-authentication-type: User
+post:
+  tags:
+  - Client
+  summary: Create custom user course
+  description: |
+    Create custom user course
+  security:
+    - bearerAuth: []
+  parameters:
+    - name: key
+      in: path
+      description: course key
+      required: true
+      style: simple
+      explode: false
+      schema:
+        type: string
+  responses:
+    200:
+      description: Success
+      content:
+        text/plain:
+          schema:
+            type: string
+            example: Success
+    400:
+      description: Bad request
+    401:
+      description: Unauthorized
+    500:
+      description: Internal error
+  x-core-function: CreateUserCourse
+  x-data-type: model.UserCourse
+  x-authentication-type: User
+delete:
```

Comment:
Let's add a PUT API to drop a user course here. The API handler should just call your storage function to mark the user course as dropped.

Reply:
write the new api, update the todo for DeleteCustomCourse